scio-avro/src/main/scala/com/spotify/scio/avro/syntax/ScioContextSyntax.scala

```diff
@@ -47,11 +48,34 @@ final class ScioContextOps(private val self: ScioContext) extends AnyVal {
     self.read(ObjectFileIO[T](path))(ObjectFileIO.ReadParam(suffix))
 
   def avroFile(path: String, schema: Schema): SCollection[GenericRecord] =
-    self.read(GenericRecordIO(path, schema))(AvroIO.ReadParam())
+    avroFile(path, schema, GenericRecordIO.ReadParam.DefaultSuffix)
 
   // overloaded API. We can't use default params
-  def avroFile(path: String, schema: Schema, suffix: String): SCollection[GenericRecord] =
-    self.read(GenericRecordIO(path, schema))(AvroIO.ReadParam(suffix))
+  def avroFile(
```

Comment:
In the ideal situation we should have `avroGenericFile` and `avroSpecificFile` to avoid those method overload

Reply:
Agree that the overloading in these methods has always been confusing -- particularly with the `schema` param since it's only required for Generic reads. On the other hand, it's probably the most commonly used Scio API and migration pain would be high, I think... maybe we can think about this for 0.15?

Reply:
sure. nothing urgent. Just wanted to raise this since adding another new parameter to the API will be a pain.